### PR TITLE
Implement multi domain p methods

### PR DESCRIPTION
**Pull request by rgraff**
Was easier to switch to yield than I thought while rebasing.

* Adds dp_, dnp_, Dn, Dnp_ methods
* Makes blocks work uniformly and everywhere
* Readme updates as appropriate.

Should be able to merge this as individual commits, but it makes more sense IMO as a single PR given the required refactoring at the end.

### CODE REVIEW

File: lib/fast_gettext/translation.rb [hunk 1]
```diff
@@ -128,44 +122,98 @@ def ds_(domain, key, separator=nil, &block)
...
+
     def dns_(domain, *keys, &block)
       _in_domain domain do
         FastGettext::Translation.ns_(*keys, &block)
       end
     end
 
+    def dnp_(domain, context, key, *args)
+      _in_domain domain do
+        result = FastGettext::Translation.np_(context, key, *args){nil}
```
Comment (grosser) [inline lib/fast_gettext/translation.rb]: should pass the block here instead of extra yield logic below ?

File: lib/fast_gettext/translation.rb [hunk 2]
```diff
@@ -128,44 +122,98 @@ def ds_(domain, key, separator=nil, &block)
...
+      FastGettext.translation_repositories.each_key do |domain|
+        result = FastGettext::TranslationMultidomain.dnp_(domain, context, *keys){ nil }
+        return result unless result.nil?
+      end
+      block_given? ? yield : FastGettext.p_(context, FastGettext::PluralizationHelper.fallback(*keys))
+    end
+  end
+
+  module PluralizationHelper
+    extend self
```
Comment (grosser) [inline lib/fast_gettext/translation.rb]: maybe better to do `class << self` or `def self.fallback` ?

File: lib/fast_gettext/translation.rb [hunk 3]
```diff
@@ -71,18 +60,23 @@ def Nn_(*keys)
...
+
+      FastGettext::PluralizationHelper.fallback(*keys).split(NAMESPACE_SEPARATOR).last
     end
     alias :nsgettext :ns_
 
-    def np_(context, key, *args)
-      options = (args.last.is_a? Hash) ? args.pop : {}
-      nargs = ["#{context}#{options[:separator]||CONTEXT_SEPARATOR}#{key}"] + args
-      n_(*nargs){nil} or (block_given? ? yield : key)
+    def np_(context, *keys)
```
Comment (grosser) [inline lib/fast_gettext/translation.rb]: a keyword-arg `separator: nil` would be nice here to get rid of the options hash logic

**Comment by grosser** [comment]
overall looks about right :)

File: lib/fast_gettext/translation.rb (hunk 1 above)
Comment (rgraff) [reply lib/fast_gettext/translation.rb]: Trying to avoid passing block.

**Comment by rgraff** [comment]
Seems good to merge now

**Comment by grosser** [comment]
too late!

**Comment by grosser** [comment]
are you done now ?

**Comment by rgraff** [comment]
I'm done here. I might have a PR for gettext_i18n_rails later in the week.

Thanks for great gems.

**Comment by grosser** [comment]
np, just scratching my own itches :D

**Comment by grosser** [comment]
I'll double check the logic a bit and then release later today

**Pull request merged**